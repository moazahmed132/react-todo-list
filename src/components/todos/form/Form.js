import React from 'react'

const Form = (props) => {
  return (
    <div>
      <form className="form-container" onSubmit={props.addTodo}>
        <input
          className="form-input"
          type="text"
          onChange={props.updateTodo}
          value={props.current}
        />
        <button type="submit" className="myButton">Add Todo</button>
      </form>
    </div>
  )
}

export default Form;
