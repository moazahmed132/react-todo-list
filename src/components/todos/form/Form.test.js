import React from 'react';
import Form from './Form';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from 'sinon';


describe('InputText Component', () => {
  let component;
  let props;
  let keywords = 'this is some keyword';

  Enzyme.configure({
    adapter: new Adapter()
  });

  beforeEach(() => {
    props = {
      onChange: jest.fn()
    }

    component = shallow(
      <Form
        value={keywords}
        onChange={props.onChange} />
    );
  })

  it('Should render the right way', () => {
    expect(component).toMatchSnapshot();
  });

  it('Should render the button text the right way', () => {
    expect(component.find('button').text()).toBe('Add Todo')
  });


  it('should call the onChange function', () => {
    let mountedComponent = mount(
      <Form
        value={keywords}
        onChange={props.onChange} />
    );

    mountedComponent.find('input').simulate('change');

    expect(props.onChange).toHaveBeenCalled();
  });

  it('should change the value of the input onChange', () => {
    keywords = 'shawky';
    let mountedComponent = mount(
      <Form
        value={keywords}
        onChange={props.onChange} />
    );

    mountedComponent.find('input').simulate('change');

    expect(mountedComponent.find('input').prop('value')).toBe('shawky');
  })
  it('simulates click events', () => {
    const onButtonClick = sinon.spy();
    const wrapper = shallow(<Form onButtonClick={onButtonClick} />);
    wrapper.find('button').simulate('click');
    expect(onButtonClick).to.have.property('addTodo', 1);
  });
})
