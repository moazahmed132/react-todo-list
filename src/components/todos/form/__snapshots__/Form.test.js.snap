// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component Should render the right way 1`] = `ShallowWrapper {}`;

exports[`InputText Component Should render the right way 1`] = `ShallowWrapper {}`;
